### ISSUE TITLE
[Feature request] Add support for syncing with Nextcloud News

### ISSUE THREAD

**Issue by protist** [akrennmair/newsbeuter#566]
**Newsbeuter version (copy from `newsbeuter -v`)**:
```
newsbeuter 2.9 - http://www.newsbeuter.org/
Copyright (C) 2006-2010 Andreas Krennmair

newsbeuter is free software and licensed under the MIT/X Consortium License.
Type `newsbeuter -vv' for more information.

Compilation date/time: Mar  8 2017 21:56:57
System: Linux 4.11.3-1-ARCH (x86_64)
Compiler: g++ 6.3.1 20170306
ncurses: ncurses 6.0.20170527 (compiled with 6.0)
libcurl: libcurl/7.54.0 OpenSSL/1.1.0f zlib/1.2.11 libpsl/0.17.0 (+libicu/59.1) libssh2/1.8.0 nghttp2/1.23.1 (compiled with 7.53.1)
SQLite: 3.19.3 (compiled with 3.17.0)
libxml2: compiled with 2.9.4
```
Steps to reproduce the issue:

1. Attempt to sync with Nextcloud News

**Other info you think is relevant**:
It would be most excellent if Newsbeuter could sync with [Nextcloud News](https://github.com/nextcloud/news). Nextcloud is the replacement to OwnCloud; the "News" app is a web-based feed aggregator similar to Google Reader, etc. There are over a dozen existing clients, ~3 of them Linux, but none of them command-line based. Adding Nextcloud News support to Newsbeuter would be very useful! Thank you.

**Comment by Minoru** [comment]
Do they support ownCloud News API? We already have ownCloud News support in master.

"Legacy API"seem to be it, but it's hard to check from the phone.

--
Regards,
Alexander Batischev

**Comment by Minoru** [comment]
Ok, I checked and [Legacy API](https://github.com/nextcloud/news/blob/master/docs/externalapi/Legacy.md) is indeed the one used by ownCloud News; we already have an implementation for that.

**Comment by protist** [comment]
Oh great! Thanks for that. However, I can't seem to get it working. I have the following in my newsbeuter config.
```
urls-source "ocnews"
ocnews-login "protist"
ocnews-passwordeval "python -m keyring get owncloud protist"
ocnews-url "https://protist.dynu.com/nextcloud"
```

However I can't connect.
```
$ newsbeauter -l 6
Starting newsbeuter 2.10-d1c1...
Loading configuration...done.
Opening cache...done.
Loading URLs from ownCloud News...Authentication failed.
```

I also tried hard-coding the password with `ocnews-password`, but that had the same error.

**Comment by protist** [comment]
In my previous comment, I updated to master, then changed my config for ocnews, and got the above errors. I now reverted my config to syncing with tt-rss, and I get the `Authentication failed` errors again. I reverted to newsbeuter 2.9, and tt-rss authentication works again.

**Comment by Minoru** [comment]
Do you have a self-signed certificate there? The domain doesn't resolve for me at the moment, so I can't check myself.

**Comment by Minoru** [comment]
Actually, I shouldn't be so lazy, and just state my hypothesis. So: in `master` we've added `ssl-verifyhost` and `ssl-verifypeer` settings, both set to "yes" by default. [Here's the docs](https://github.com/akrennmair/newsbeuter/blob/d1c116d570f13276c21b62c215c3c3f8da69d576/doc/configcommands.dsv#L89).

They don't affect TT-RSS and Newsblur (see #537), but should work fine for ownCloud News (and hence nextCloud News). If your certificate is indeed self-signed, try setting both options to "no" and trying with nextCloud again.

**Comment by protist** [comment]
> Do you have a self-signed certificate there?

Yes, I do.

> The domain doesn't resolve for me at the moment, so I can't check myself.

Oh, sorry, the domain name from my config was just dummy text to maintain privacy.

>They don't affect TT-RSS…

I actually found that I *did* need to set `ssl-verifypeer no` to get even tt-rss to work.

Having said that, this also worked well for Nextcloud too. (Again, I only needed the one flag for my setup.) Thank you!

However, with the master version, Newsbeuter is ridiculously slow with some commands with Nexcloud. I tried marking articles read with `N`. For ten articles, it averaged 1.6 seconds each. I re-tested with tt-rss on the same newsbeuter version, and it is near instantaneous.
